nixos/modules/services/networking/thelounge.nix

```diff
@@ -88,16 +140,53 @@ in
     systemd.services.thelounge = {
       description = "The Lounge web IRC client";
       wantedBy = [ "multi-user.target" ];
-      preStart = "ln -sf ${pkgs.writeText "config.js" configJsData} ${dataDir}/config.js";
+      preStart = ''
+        set -o errexit -o pipefail -o nounset -o errtrace
+        shopt -s inherit_errexit
+
+        ln -sf ${pkgs.writeText "config.js" configJsData} ${dataDir}/config.js
+
+        mkdir ${dataDir}/users
+
+        ${concatStrings (mapAttrsToList (user: options:
+          let
+            path = "${dataDir}/users/${user}.json";
+          in
+          ''
+            if [ ! -e ${path} ]; then
+              ${if options.hashedPassword != null then ''
+                PASSWORD=${escapeShellArg options.hashedPassword}
+              '' else ''
+                PASSWORD="$(<${escapeShellArg options.hashedPasswordFile})"
+                export PASSWORD
+              ''}
```

Comment:
```suggestion
              ''}
              export PASSWORD
```

Reply:
I understand why this change is done, so why did the `hashedPassword` option work without exporting it?

Reply:
It shouldn't. `jq` won't complain about missing environment variables, though - it just reads them as `null`.

Reply:
Weird, it definitely works fine without being exported; I've confirmed this by checking the `jq` output. Do you have any clue why this would be happening?

Reply:
You're right, it worked because `jane` was put before `john`, thus exporting the variable.